### DISCUSSION TITLE
Flatpak or AppImage version planned?

### DISCUSSION THREAD

**Discussion by hyuri** [friction2d/friction#70] [General]
Hi!

Any plans to offer the app as a Flatpak, and even upload it to Flathub? It would make it easier for more people to try the app, especially since Flathub is seeing millions of downloads every month (https://flathub.org/statistics).

Or at least AppImage?

These contained, non-root formats at least give us some peace of mind knowing they won't intentionally or accidentally destroy our system. With .debs, there's always a fear it may fail to install and leave something half-installed that breaks the system. And then there's the need to maintain a new version for every OS release. With Flatpaks and AppImages, you get one app version for all OS versions.

**Comment by rodlie** [comment]
Not really ...

### AppImage

Is a big hack, maintained by someone that don't know how to deploy software on Linux.

It's also too much work to maintain (all the depends for Friction will need to be built and maintained), so big no from me.

### Flatpak

In general I don't like Flatpak (concept or implementation), and it will create additional (sandbox) issues that's unrelated to the application. But I'm not a 100% against it, so maybe ...

---

> With .debs, there's always a fear it may fail to install and leave something half-installed that breaks the system

Don't see the problem. The Friction package will not break your system (look at the content of the package). Use it or build it yourself ;)

**Comment by dadoprom** [comment]
.debs are perfect for me :)

**Comment by hyuri** [comment]
Earlier this year, a Fade In Pro .deb install failed, left broken packages
and I couldn't install it anymore. I had to reinstall the system.

You're lucky debs have been working for you, but you can easily find many
reports on the internet of people with half-installed packages breaking
their systems. None of theses issues would ever happen to a flatpak because
they are contained, and allow you to easily install afresh without touching
anything else.

Not to mention they don't require root privileges to run.

Of course they are not perfect but, for most users, it's the most
hassle-free option, and, if you publish it on Flathub, it'll even show up
by default on their Flathub-enabled app stores like Gnome Software etc. in
many systems.

So I think it makes sense and the trade-offs are worth it, especially if
you want more users to try the app.

**Comment by rodlie** [comment]
You can always just extract the content of the package and run it from whatever.

Will probably do an flatpak at some point, but there are more important things to consider at the moment.

Anyway, created an isssue https://github.com/friction2d/friction/issues/72, will close this :)
